datafusion/optimizer/src/simplify_expressions/simplify_exprs.rs

```diff
@@ -93,16 +93,33 @@ impl SimplifyExpressions {
             .map(|input| Self::optimize_internal(input, execution_props))
             .collect::<Result<Vec<_>>>()?;
 
-        let expr = plan
-            .expressions()
-            .into_iter()
-            .map(|e| {
-                // TODO: unify with `rewrite_preserving_name`
-                let original_name = e.name_for_alias()?;
-                let new_e = simplifier.simplify(e)?;
-                new_e.alias_if_changed(original_name)
-            })
-            .collect::<Result<Vec<_>>>()?;
+        let expr = match plan {
+            // Canonicalize step won't reorder expressions in a Join on clause.
+            // The left and right expressions in a Join on clause are not commutative.
```

Comment:
```suggestion
            // Canonicalize step won't reorder expressions in a Join on clause.
            // The left and right expressions in a Join on clause are not commutative,
            // since the order of the columns must match the order of the children.
```

Just an extra bit of clarification